### ISSUE TITLE
Video UNet not learning?

### ISSUE THREAD

**Issue by julian-q** [buoyancy99/diffusion-forcing#13]
![image](https://github.com/user-attachments/assets/36c5144b-25c4-4d6e-a236-fe48929827dd)

Hi, thanks for the nice repo! I'm trying to follow the instructions for training a video model on Minecraft, and it looks like the loss is staying stagnant for 60k steps.

Command:
`python -m main +name=minecraft_unet algorithm=df_video dataset=video_minecraft`

Is this normal? Any tips much appreciated!!

**Comment by buoyancy99** [comment]
Hello, 

How does the training_vis images look? Do they look legit? 

I am wondering whether you finished downloading the entire dataset, which can take a couple days, or did you accidentally used the mini subset shipped with the checkpoint file?

Some people trained with the mini subset and as a result, it's only training with like 10 videos and overfit.

**Comment by julian-q** [comment]
Here's the latest training_vis/video_2 from wandb - it looks pretty noisy:
![video_2_6086_b0aaf937f038141a0728](https://github.com/user-attachments/assets/acc6c960-5d2b-4381-9692-eaf0544cdc81)

I think I have the whole dataset because I symlinked the dataset I previously downloaded with the TECO bash script.

Here's my [`config.yaml`](https://gist.github.com/julian-q/048d6f792767d94516369a2df5587f0d). I think the one modification I made was the batch size. Maybe I should retry on a machine with more memory so that I can use the original config.

Update: Oh, oops, it turns out I was not training on the whole dataset because I was still using the `metadata.json` from the sample dataset. To create a `metadata.json` for my existing TECO dataset, I wrote [this script](https://gist.github.com/julian-q/eb19e4c034b731adeee5d7aa6ececf49). I'll train again with this metadata file and see how it goes!

**Comment by julian-q** [comment]
Even after training with the original config on the whole dataset, it still looks stagnant 🤔
<img width="446" alt="Screenshot 2024-08-15 at 11 57 51 AM" src="https://github.com/user-attachments/assets/35caf1b8-019a-4c22-aaa7-b9609f6e038c">

And the training vis is still noisy:
![video_2_4993_bd4aaf968e0eca8147c1](https://github.com/user-attachments/assets/1ce8ffdb-6360-4260-a4cf-451534abd25e)

**Comment by julian-q** [comment]
I don't think it's a problem with the dataset, since when I switch to the `paper` branch and train the RNN-based video model, the loss goes down just fine

<img width="353" alt="Screenshot 2024-08-15 at 1 44 58 PM" src="https://github.com/user-attachments/assets/3b61a5e3-4c80-4095-9dcf-2aeab9044276">

"minecraft_video" is on the `paper` branch, "minecraft_unet_transformer" is on the `main` branch, same data for both runs

**Comment by julian-q** [comment]
Okay, I tried cloning the [`diffusion-forcing-transformer`](https://github.com/kwsong0113/diffusion-forcing-transformer/tree/f8a8e52630726c9e86e865d0a523981f34de054f) repo for the original implementation of the transformer UNet to see if there's a difference. I started training using [this command](https://github.com/kwsong0113/diffusion-forcing-transformer/blob/f8a8e52630726c9e86e865d0a523981f34de054f/README.md?plain=1#L53), and I now see the loss going down:

<img width="445" alt="Screenshot 2024-08-15 at 2 54 08 PM" src="https://github.com/user-attachments/assets/701fc194-2f1e-4b07-a5b0-bc8ff6c2f868">

After 9k steps the training vis looks good too:
![video_2_427_871367e5c96e2fa56b1a](https://github.com/user-attachments/assets/f423a2f9-8463-43d8-8438-5bc445c62630)

Will try to see what the discrepancy is between `main` and `diffusion-forcing-transformer` ..

**Comment by buoyancy99** [comment]
Hi Julian I will immediately start to debug this - Right now the v1.5 video diffusion is supposed to be exactly the same as diffusion-forcing-transformer

**Comment by buoyancy99** [comment]
Thank you for pointing this out, I found the bug. Can you pull this repo again?

The bug fix is [this](https://github.com/buoyancy99/diffusion-forcing/commit/17620312ce5238afe1fc57572a733f595c40fcef)

**Comment by julian-q** [comment]
Haha, nice catch:) Loss is going down now on `main`! Thank you 🙏
